OsmAnd/src/net/osmand/plus/server/OsmAndHttpServer.java

```diff
@@ -0,0 +1,126 @@
+package net.osmand.plus.server;
+
+import android.util.Log;
+import android.webkit.MimeTypeMap;
+import fi.iki.elonen.NanoHTTPD;
+import net.osmand.PlatformUtil;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.server.endpoints.TileEndpoint;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OsmAndHttpServer extends NanoHTTPD {
+	private static final String FOLDER_NAME = "server";
+	private static final org.apache.commons.logging.Log LOG = PlatformUtil.getLog(OsmAndHttpServer.class);
+	public static final int PORT = 24990;
+	public static String HOSTNAME = "0.0.0.0";
```

Comment:
static and not final variable?

Reply:
Yes, The user can change HOSTNAME for the server that's why it is not final. But the user is allowed to have only one instance of server that's why it is static.